security-admin/src/main/java/org/apache/ranger/rest/XUserREST.java

```diff
@@ -165,6 +166,10 @@ public class XUserREST {
 	
 	static final Logger logger = LoggerFactory.getLogger(XUserMgr.class);
 
+	static volatile boolean usersDeletionInProgress = false;
```

Comment:
Use of static variables wouldn't work in HA deployments, as subsequent request might be going to a different instance. Why is it critical to avoid multiple simultaneous calls?

Reply:
Sure, makes sense. Sometimes these calls might take long time to get a response, the idea was to get a response indicating the progress(like 30% deletion complete or deletion in progress check back later) in the interim via another simultaneous call.